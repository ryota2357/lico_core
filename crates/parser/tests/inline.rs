//! This file is generated by `cargo codegen parser`, do not edit by hand.
#![cfg_attr(rustfmt, rustfmt::skip)]

mod utils;
const GROUP: &str = "inline";

#[test]
fn expr_literals() {
    let source = include_str!("../test_data/inline/expr_literals.lico");
    let snapshot = utils::make_snapshot("expr_literals", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn param_list() {
    let source = include_str!("../test_data/inline/param_list.lico");
    let snapshot = utils::make_snapshot("param_list", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn attr_stmt() {
    let source = include_str!("../test_data/inline/attr_stmt.lico");
    let snapshot = utils::make_snapshot("attr_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn break_stmt() {
    let source = include_str!("../test_data/inline/break_stmt.lico");
    let snapshot = utils::make_snapshot("break_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn continue_stmt() {
    let source = include_str!("../test_data/inline/continue_stmt.lico");
    let snapshot = utils::make_snapshot("continue_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn func() {
    let source = include_str!("../test_data/inline/func.lico");
    let snapshot = utils::make_snapshot("func", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn return_stmt() {
    let source = include_str!("../test_data/inline/return_stmt.lico");
    let snapshot = utils::make_snapshot("return_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn var_stmt() {
    let source = include_str!("../test_data/inline/var_stmt.lico");
    let snapshot = utils::make_snapshot("var_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn while_stmt() {
    let source = include_str!("../test_data/inline/while_stmt.lico");
    let snapshot = utils::make_snapshot("while_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
