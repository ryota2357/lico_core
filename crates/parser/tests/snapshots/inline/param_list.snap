---
source: crates/parser/tests/inline.rs
snapshot_kind: text
---
# param_list

## Input

```
func f(a) end
func f(a, b) end
func f(a, b,) end
```

## CST

```
SOURCE_FILE@0..48
  FUNC_STMT@0..13
    FUNC_KW@0..4 "func"
    WHITESPACE@4..5 " "
    NAME@5..6
      IDENT@5..6 "f"
    PARAM_LIST@6..9
      OPEN_PAREN@6..7 "("
      PARAM@7..8
        NAME@7..8
          IDENT@7..8 "a"
      CLOSE_PAREN@8..9 ")"
    WHITESPACE@9..10 " "
    END_KW@10..13 "end"
  WHITESPACE@13..14 "\n"
  FUNC_STMT@14..30
    FUNC_KW@14..18 "func"
    WHITESPACE@18..19 " "
    NAME@19..20
      IDENT@19..20 "f"
    PARAM_LIST@20..26
      OPEN_PAREN@20..21 "("
      PARAM@21..22
        NAME@21..22
          IDENT@21..22 "a"
      COMMA@22..23 ","
      WHITESPACE@23..24 " "
      PARAM@24..25
        NAME@24..25
          IDENT@24..25 "b"
      CLOSE_PAREN@25..26 ")"
    WHITESPACE@26..27 " "
    END_KW@27..30 "end"
  WHITESPACE@30..31 "\n"
  FUNC_STMT@31..48
    FUNC_KW@31..35 "func"
    WHITESPACE@35..36 " "
    NAME@36..37
      IDENT@36..37 "f"
    PARAM_LIST@37..44
      OPEN_PAREN@37..38 "("
      PARAM@38..39
        NAME@38..39
          IDENT@38..39 "a"
      COMMA@39..40 ","
      WHITESPACE@40..41 " "
      PARAM@41..42
        NAME@41..42
          IDENT@41..42 "b"
      COMMA@42..43 ","
      CLOSE_PAREN@43..44 ")"
    WHITESPACE@44..45 " "
    END_KW@45..48 "end"
```

## AST

```
SourceFile {
    shebang: none,
    stmt: [
        FuncStmt {
            func_token: FUNC_KW@0..4 "func",
            name: Name {
                ident: IDENT@5..6 "f",
            },
            param_list: ParamList {
                open_paren_token: OPEN_PAREN@6..7 "(",
                params: [
                    Param {
                        name: Name {
                            ident: IDENT@7..8 "a",
                        },
                    },
                ],
                close_paren_token: CLOSE_PAREN@8..9 ")",
            },
            statements: [],
            end_token: END_KW@10..13 "end",
        },
        FuncStmt {
            func_token: FUNC_KW@14..18 "func",
            name: Name {
                ident: IDENT@19..20 "f",
            },
            param_list: ParamList {
                open_paren_token: OPEN_PAREN@20..21 "(",
                params: [
                    Param {
                        name: Name {
                            ident: IDENT@21..22 "a",
                        },
                    },
                    Param {
                        name: Name {
                            ident: IDENT@24..25 "b",
                        },
                    },
                ],
                close_paren_token: CLOSE_PAREN@25..26 ")",
            },
            statements: [],
            end_token: END_KW@27..30 "end",
        },
        FuncStmt {
            func_token: FUNC_KW@31..35 "func",
            name: Name {
                ident: IDENT@36..37 "f",
            },
            param_list: ParamList {
                open_paren_token: OPEN_PAREN@37..38 "(",
                params: [
                    Param {
                        name: Name {
                            ident: IDENT@38..39 "a",
                        },
                    },
                    Param {
                        name: Name {
                            ident: IDENT@41..42 "b",
                        },
                    },
                ],
                close_paren_token: CLOSE_PAREN@43..44 ")",
            },
            statements: [],
            end_token: END_KW@45..48 "end",
        },
    ],
}
```

vim:ft=markdown
